src/mame/f256/tiny_vicky.cpp

```diff
@@ -0,0 +1,613 @@
+                uint16_t sol_line = m_iopage0_ptr[0x1018] + ((m_iopage0_ptr[0x101A] & 0xF) << 8);
+                uint32_t *row = topleft + y * 800;
+                if ((sol_reg & 1) != 0 && y == sol_line)
+                {
+                    // raise an interrupt
+                    m_sol_irq_handler(1);
+                }
+                // border color can change during painting the screen
+                if (display_border)
+                {
+                    if (!enable_gamma)
+                    {
+                        border_color = rgb_t(m_iopage0_ptr[0x1007], m_iopage0_ptr[0x1006], m_iopage0_ptr[0x1005]);
+                    }
+                    else
+                    {
+                        border_color = rgb_t(m_iopage0_ptr[0x800 + m_iopage0_ptr[0x1007]],
+                                             m_iopage0_ptr[0x400 + m_iopage0_ptr[0x1006]],
+                                             m_iopage0_ptr[m_iopage0_ptr[0x1005]]);
+                    }
+                }
+
+                if (y < border_y || y >= lines - border_y)
+                {
+                    for (int x = 0; x < 640; x++)
+                    {
+                        row[x] = border_color;
+                    }
+                }
+                else
+                {
+                    rgb_t background_color = rgb_t();
+                    // only draw the even lines
+                    if (enable_graphics && y % 2 == 0)
+                    {
+                        if (!enable_gamma)
+                        {
+                            background_color = rgb_t(m_iopage0_ptr[0x100F], m_iopage0_ptr[0x100E], m_iopage0_ptr[0x100D]);
+                        }
+                        else
+                        {
+                            background_color = rgb_t(m_iopage0_ptr[0x800 + m_iopage0_ptr[0x100F]],
+                                                    m_iopage0_ptr[0x400 + m_iopage0_ptr[0x100E]],
+                                                    m_iopage0_ptr[m_iopage0_ptr[0x100D]]);
+                        }
+                    }
+                    // draw the border or background
+                    if (y % 2 == 0)
+                    {
+                        for (int x = 0; x < 640; x++)
+                        {
+                            row[x] = x < border_x || x >= 640 - border_x ? border_color : background_color;
+                        }
+                        memcpy(row+800,row, 640*4);
+                    }
+
+                    if (enable_graphics)
+                    {
+                        if (y % 2 == 0)
+                        {
+                            // Tiny Vicky Layers for Bitmaps, Tilemaps and sprites
+                            uint8_t LayerMgr0 = m_iopage0_ptr[0xD002 - 0xC000] & 0x7;
+                            uint8_t LayerMgr1 = m_iopage0_ptr[0xD002 - 0xC000] >> 4;
+                            uint8_t LayerMgr2 = m_iopage0_ptr[0xD003 - 0xC000] & 0x7;
```

Comment:
Please use `snake_case` for local variables.  We tend to reserve `LlamaCase` for template parameters.

Also, please consistently use lowercase hex digits.